src/unpack.rs

```diff
@@ -269,3 +261,18 @@ fn worker_proc(in_file: &mut File, version: u32, variant: Variant, encryption_ke
 
     Ok(())
 }
+
+fn decrypt_entry(buffer: &mut Vec<u8>, record: &Record, encryption_key: Option<Vec<u8>>, size: usize) -> Result<()> {
+    if record.encrypted() {
+        if let Some(key) = encryption_key {
+            decrypt(buffer, key);
+            // Trim padded bytes from undersized encryption blocks
+            *buffer = buffer[..size].to_vec();
```

Comment:
That seems inefficient (allocating a new buffer, freeing the old). Why not just `buffer.truncate(size);`?

Reply:
Done